openraft/src/membership/membership.rs

```diff
@@ -106,6 +106,21 @@ impl Membership {
         }
     }
 
+    #[must_use]
+    pub fn add_learner_set(&self, add_learners: BTreeSet<NodeId>) -> Self {
+        let mut learners = self.learners.clone();
+        for node_id in add_learners {
+            learners.insert(node_id);
+        }
+        let configs = self.configs.clone();
```

Comment:
What about adding an internal method to clean up node ids from `learners` already in `configs`.